src/node/markdown/markdown.ts

```diff
@@ -238,7 +240,7 @@ export const createMarkdownRenderer = async (
     slugify,
     permalink: anchorPlugin.permalink.linkInsideHeader({
       symbol: '&ZeroWidthSpace;',
-      renderAttrs: (slug, state) => {
+      renderAttrs: (slug, state: StateCore) => {
```

Comment:
if we are going down the patch route, maybe patch this as well

node_modules/markdown-it-anchor/types/index.d.ts

```ts
import MarkdownIt from 'markdown-it';
import Token from 'markdown-it/lib/token.mjs'
import State from 'markdown-it/lib/rules_core/state_core.mjs';
```

this should fix those unresolved imports in dts build too

Reply:
Patched `markdown-it-anchor` and reverted the change here